upload_vehicle_data.py

```diff
@@ -0,0 +1,729 @@
+            for rosbag in rosbag_list
+        ]
+        total_size_gb = sum(rosbag_sizes)
+        estimated_time = get_estimated_upload_time(
+            total_size_gb, bandwidth_mbps, rosbag_sizes
+        )
+
+        if estimated_time < 1:
+            estimated_time_str = f'{estimated_time * 60:.2f} minutes'
+        else:
+            estimated_time_str = f'{estimated_time:.2f} hours'
+
+        logger.info(
+            f'Found {len(rosbag_list)} files to upload with total size '
+            f'{total_size_gb:.2f} GB from {remote_directory}.'
+        )
+        logger.info(
+            f'Estimated total time (including compression) '
+            f'is at least: {estimated_time_str}.'
+        )
+
+        successfully_uploaded_files = []
+
+        # Use ThreadPoolExecutor for parallel processing
+        with ThreadPoolExecutor(
+            max_workers=config['parallel_processes']
+        ) as executor:
+            futures = [
+                executor.submit(
+                    compress_and_transfer_rosbag,
+                    logger,
+                    remote_user,
+                    remote_ip,
+                    rosbag,
+                    remote_directory,
+                    cloud_upload_directory,
+                    config['mcap_path'],
+                    config['upload_attempts'],
+                    base_remote_directory,
+                )
+                for rosbag in rosbag_list
+            ]
+
+            for future in futures:
+                result = future.result()
+                if result:
+                    successfully_uploaded_files.append(result)
+
+        if config['clean_up']:
+            # Only delete rosbag files that were successfully uploaded
+            for rosbag in successfully_uploaded_files:
+                delete_remote_file(logger, remote_user, remote_ip, rosbag)
+
+    finally:
+        # Delete the remote temporary directory
+        delete_remote_temp_directory(
+            logger, remote_user, remote_ip, remote_directory
+        )
+
+
+def main(config, debug):
+    """Automate the upload of rosbags."""
+    logger = setup_logging(debug)
+    remote_user = config['remote_user']
+    remote_ip = config['remote_ip']
+    base_remote_directory = config['remote_directory']
+    cloud_upload_directory = config['cloud_upload_directory']
+    directory_depth = config.get(
+        'directory_depth', 1
+    )  # Default to 1 for flat structure
+    logger.info('Starting rosbag upload process.')
+
+    # Measure bandwidth once at the start
+    bandwidth_mbps = measure_bandwidth(logger, remote_ip, remote_user)
+    if bandwidth_mbps is None:
+        logger.error('Could not measure bandwidth. Exiting.')
+        return
+    # Get all subdirectories in the base remote directory
+    subdirectories = list_remote_directories(
+        logger, remote_user, remote_ip, base_remote_directory, directory_depth
+    )
+    logger.debug(f'Found subdirectories: {subdirectories}')

```

Comment:
This line creates a cluttered log in the terminal, lets do a change:

```suggestion
    logger.debug(f'Rosbags subdirectories found: {len(subdirectories)}')

```

Reply:
Changed